spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -312,8 +314,15 @@ protected MessageHandler createProducerMessageHandler(final ProducerDestination
 			checkConnectionFactoryIsErrorCapable();
 			endpoint.setReturnChannel(errorChannel);
 			endpoint.setConfirmNackChannel(errorChannel);
-			endpoint.setConfirmAckChannel(getApplicationContext().getBean(
-					IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME, MessageChannel.class));
+			String ackChannelBeanName = StringUtils.hasText(extendedProperties.getConfirmAckChannel())
+					? extendedProperties.getConfirmAckChannel()
+					: IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME;
+			if (!ackChannelBeanName.equals(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)
+					&& !getApplicationContext().containsBean(ackChannelBeanName)) {
+				GenericApplicationContext context = (GenericApplicationContext) getApplicationContext();
+				context.registerBean(ackChannelBeanName, DirectChannel.class, () -> new DirectChannel());
+			}
+			endpoint.setConfirmAckChannel(getApplicationContext().getBean(ackChannelBeanName, MessageChannel.class));
```

Comment:
Maybe let's use a `setConfirmAckChannelName()` instead to avoid some potential eager initialization?